django_postmark_utils/models.py

```diff
@@ -1,3 +1,269 @@
+        _("Pickled object"),
+        help_text=_("Pickled message object")
+    )
+    message_id = models.CharField(
+        _("Message ID"),
+        max_length=255,
+        unique=True,
+        help_text=_("The 'Message-ID' header field of the message")
+    )
+    date = models.DateTimeField(
+        _("Date"),
+        help_text=_("The 'Date' header field of the message")
+    )
+    subject = models.CharField(
+        _("Subject"),
+        max_length=255,
+        blank=True,
+        help_text=_("The 'Subject' header field of the message")
+    )
+    from_email = models.CharField(
+        _("Sender email address"),
+        max_length=255,
+        blank=True,
+        help_text=_("The 'From' header field of the message")
+    )
+    to_emails = models.TextField(
+        _("Recipient email addresses"),
+        blank=True,
+        help_text=_("The 'To' field of the message, with email addresses "
+                    "separated by commas")
+    )
+    cc_emails = models.TextField(
+        _("Cc'd recipient email addresses"),
+        blank=True,
+        help_text=_("The 'Cc' field of the message, with email addresses "
+                    "separated by commas")
+    )
+    bcc_emails = models.TextField(
+        _("Bcc'd recipient email addresses"),
+        blank=True,
+        help_text=_("The 'Bcc' field of the message, with email addresses "
+                    "separated by commas")
+    )
+    delivery_status = models.IntegerField(
+        _("Delivery status"),
+        help_text=_("The delivery status of the message"),
+    )
+    has_been_resent = models.BooleanField(
+        _("Has been resent"),
+        default=False,
+        help_text=_("If the message has been resent, after an exception "
+                    "prevented this from happening the first time round")
+    )
+    delivery_submission_date = models.DateTimeField(
+        _("Delivery-submission date"),
+        null=True,
+        blank=True,
+        help_text=_("When the message was submitted for delivery")
+    )
+    delivery_message_id = models.CharField(
+        _("Delivery message ID"),
+        max_length=255,
+        unique=True,
+        blank=True,
+        help_text=_("The 'Message-ID' header field of the message, as set by "
+                    "Postmark")
+    )
+    delivery_error_code = models.IntegerField(
+        _("Delivery error code"),
+        null=True,
+        blank=True,
+        help_text=_("The delivery error code of the message, as specified here: "
+                    "<https://postmarkapp.com/developer/api/overview#error-codes>"),
+    )
+    delivery_message = models.CharField(
+        _("Delivery message"),
+        max_length=255,
+        blank=True,
+        help_text=_("The response message from Postmark")
+    )
+
+    class Meta:
+        verbose_name = _("message")
+        verbose_name_plural = _("messages")
+        ordering = ['-date']
+
+    def update_delivery_status(self):
```

Comment:
This whole method gives me an unspecific queasy feeling... there are too many repeated lines and branching conditionals statements?